kit/Kit.cpp

```diff
@@ -2822,15 +2845,24 @@ int KitSocketPoll::kitPoll(int timeoutMicroS)
             if (_document && _document->needsQuickPoll())
                 realTimeout = 0;
 
-            if (poll(std::chrono::microseconds(realTimeout)) <= 0)
+            const int rc = poll(std::chrono::microseconds(realTimeout));
```

Comment:
Not convinced by this. What 'idle' are we detecting here. We can tell when the core main-loop is idle from a non-zero timeout; that's fine - how do we detect our own ie. coolforkit idle-ness ? presumably with a very short poll timeout - is that what we do ? confused here. Looks like we conclude that if we waited a long time and timed-out, we are now idle - and now we can do the things we wanted to do but were delayed; but why delay them ? 

Confused.